### PR TITLE
Allow selecting the RCTC folder

### PR DESCRIPTION
**Pull request by Gymnasiast**
This will copy all the files and apply fixes along the way. This has the advantage of making it slightly quicker (as decoding of .sea files only has to take place once, and because we avoid several if/else checks to handle two very different folder structures), as well as keeping the code to handle the different folder structure of RCTC mostly contained to a single place.

### CODE REVIEW

File: src/openrct2/config/Config.cpp [hunk 1]
```diff
@@ -706,6 +706,17 @@ namespace Config
             }
         }
 
+        //        static constexpr u8string_view rctcSearchLocations[] = {
```
Comment (Gymnasiast) [inline src/openrct2/config/Config.cpp:709]: This bit still has to be filled in with the appropriate directories where RCTC can reside on Windows and macOS. For this, I’ll need some help from Windows users.

File: src/openrct2/PlatformEnvironment.cpp [hunk 2]
```diff
@@ -219,6 +219,7 @@ const char* PlatformEnvironment::DirectoryNamesRCT2[] = {
     nullptr,       // SHADER
     nullptr,       // THEME
     "Tracks",      // TRACK
+    nullptr,       // SOURCE
```
Comment (Gymnasiast) [inline src/openrct2/PlatformEnvironment.cpp:222]:
```suggestion
    nullptr,       // RCT2_SOURCE
```

File: src/openrct2/config/Config.cpp (hunk 1 above)
Comment (karst) [reply src/openrct2/config/Config.cpp:709]: In the case of Steam on Windows this will be `Steam\steamapps\common\RollerCoaster Tycoon Classic`

File: data/language/en-GB.txt [hunk 3]
```diff
@@ -3494,7 +3494,7 @@ STR_6386    :Blizzard
 STR_6387    :Can’t lower element here…
 STR_6388    :Can’t raise element here…
 STR_6389    :Invalid clearance
-STR_6390    :OpenRCT2 needs files from the original RollerCoaster Tycoon 2 in order to work. Please select the directory where you installed RollerCoaster Tycoon 2.
+STR_6390    :OpenRCT2 needs files from the original RollerCoaster Tycoon 2 or RollerCoaster Tycoon Classic in order to work. Please select the directory where you installed RollerCoaster Tycoon 2 or RollerCoaster Tycoon Classic.
 STR_6391    :Please select your RCT2 directory
```
Comment (karst) [inline data/language/en-GB.txt:3498]: Thisone needs to be updated aswell

**Comment by karst** [comment]
I have isolated my copy of RCT2 aswell as my OpenRCT2 documents folder so I can select RCTC. I noticed the following:
* Selecting the correct path to RCTC currently doesn't actually function. The game will just close if you select that folder and you will have to redo that. 
* The selection system is an absolute pain as there's currently no ability to just input a path and just have to scroll through alot of folders every time. This can be fixed by using a different windows menu to select the folder or adding a tab where you can just input the path by hand.
* Setting the path in `config.ini` won't work either and it will just ask you again where you installed the game.

**Comment by Gymnasiast** [comment]
> (Windows testing)
> 
> I have isolated my copy of RCT2 aswell as my OpenRCT2 documents folder so I can select RCTC. I noticed the following:
> 
>     * Selecting the correct path to RCTC currently doesn't actually function. The game will just close if you select that folder and you will have to redo that.

Works for me, on Linux at least. Which folder did you select exactly? Can you give me a directory listing?

> 
>     * The selection system is an absolute pain as there's currently no ability to just input a path and just have to scroll through alot of folders every time. This can be fixed by using a different windows menu to select the folder or adding a tab where you can just input the path by hand.

That’s the standard Windows dialog, nothing we can do about and not related to this PR in any case.

> 
>     * Setting the path in `config.ini` won't work either and it will just ask you again where you installed the game.

Of course it won’t, it has to extract it first.

**Comment by karst** [comment]
> Can you give me a directory listing?

In my case this is `D:\Program Files (x86)\Steam\steamapps\common\RollerCoaster Tycoon Classic`

**Comment by Lastorder-DC** [comment]
Not works too.

Verbose log
```
OpenRCT2, v0.4.1-187-g1af28c1 (1af28c1 on refs/pull/17813/merge) provided by GitHub
Windows (x86-64)

VERBOSE: time is 2022/08/14 22:06:08
VERBOSE: registering AVX2 mask function
VERBOSE: Checking if file exists: F:\UserData\YJSoft\Documents\OpenRCT2\config.ini
VERBOSE: DIRBASE::RCT1    : E:\SteamLibrary\steamapps\common\RollerCoaster Tycoon Deluxe
VERBOSE: DIRBASE::RCT2    :
VERBOSE: DIRBASE::OPENRCT2: M:\OpenRCT2-ClassicPR\data
VERBOSE: DIRBASE::USER    : F:\UserData\YJSoft\Documents\OpenRCT2
VERBOSE: DIRBASE::CONFIG  : F:\UserData\YJSoft\Documents\OpenRCT2
VERBOSE: DIRBASE::CACHE   : F:\UserData\YJSoft\Documents\OpenRCT2
VERBOSE: SDL2 version: 2.0.22
Importing legacy shortcuts...
VERBOSE: install directory does not exist or invalid directory selected,
VERBOSE: config_find_rct2_path(...)
VERBOSE: Checking if file exists: C:\Program Files\Steam\steamapps\common\Rollercoaster Tycoon 2\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\Program Files (x86)\Steam\steamapps\common\Rollercoaster Tycoon 2\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\GOG Games\RollerCoaster Tycoon 2 Triple Thrill Pack\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\Program Files\GalaxyClient\Games\RollerCoaster Tycoon 2 Triple Thrill Pack\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\Program Files (x86)\GalaxyClient\Games\RollerCoaster Tycoon 2 Triple Thrill Pack\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\Program Files\Atari\RollerCoaster Tycoon 2\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\Program Files (x86)\Atari\RollerCoaster Tycoon 2\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\Program Files\Infogrames\RollerCoaster Tycoon 2\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\Program Files (x86)\Infogrames\RollerCoaster Tycoon 2\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\Program Files\Infogrames Interactive\RollerCoaster Tycoon 2\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\Program Files (x86)\Infogrames Interactive\RollerCoaster Tycoon 2\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: c:/program files (x86)/steam\steamapps\common\Rollercoaster Tycoon 2\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: C:\Users\YJSoft\AppData\Local\DiscordGames\RollerCoaster Tycoon 2 Triple Thrill Pack\content\Game\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: M:\OpenRCT2-ClassicPR\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: E:\SteamLibrary\steamapps\common\RollerCoaster Tycoon Classic\Data\g1.dat
VERBOSE: Checking if file exists:
VERBOSE: Checking if file exists: E:\SteamLibrary\steamapps\common\RollerCoaster Tycoon Classic\Assets\g1.dat
VERBOSE: Checking if file exists: E:\SteamLibrary\steamapps\common\RollerCoaster Tycoon Classic\Assets\g1.dat
Unable to open 'F:\UserData\YJSoft\Documents\OpenRCT2\rct2\Scenarios\adrenaline heights.sc6'
An RCT2 install directory must be specified! Please edit "game_path" in F:\UserData\YJSoft\Documents\OpenRCT2\config.ini
```

**Comment by Gymnasiast** [comment]
> > Can you give me a directory listing?
> 
> In my case this is `D:\Program Files (x86)\Steam\steamapps\common\RollerCoaster Tycoon Classic`

That’s not a listing. A listing is a list of files in that folder (and optionally subfolders).

**Comment by Umdlye** [comment]
I'm getting the same error as Lastorder-DC (Windows 10 x64). Here is the listing for the RCTC folder: https://pastebin.com/raw/kFpNE6Ly

TL;DR: all the relevant files are in the Assets subfolder.

**Comment by karst** [comment]
> > > Can you give me a directory listing?
> > 
> > 
> > In my case this is `D:\Program Files (x86)\Steam\steamapps\common\RollerCoaster Tycoon Classic`
> 
> That’s not a listing. A listing is a list of files in that folder (and optionally subfolders). I will still need that to see what’s going wrong. My local version of RCTC might not match yours.

Ah like so.

image

In the assets folder you will find all ObjectData in POB format, All soundFX and music in OGG format. All scenarios is SEA format, G1 in DAT format. Tracks in TD6 format.

**Comment by Gymnasiast** [comment]
I’m stumped. But @IntelOrca told me he wanted to implement this in a different way anyway, so I’ll just close the PR. Hopefully, this functionality will arrive soon.
